gr-uhd/lib/usrp_source_impl.cc

```diff
@@ -659,12 +660,43 @@ namespace gr {
 
       case ::uhd::rx_metadata_t::ERROR_CODE_OVERFLOW:
         _tag_now = true;
+
+        //publish message, only if someone's listening
+        if(! pmt::is_null(d_message_subscribers) )
+        {
+          pmt::pmt_t dic = pmt::make_dict();
+          dic = pmt::dict_add(dic, pmt::mp("code"), pmt::from_long(_metadata.error_code));
```

Comment:
I still don't get it (but I'm a but tired): this is explicitly in the 'case' of ERROR_CODE_OVERFLOW.

Am 6. Juni 2016 23:42:00 MESZ, schrieb mbr0wn <notifications@github.com>:
>> @@ -659,12 +660,43 @@ namespace gr {
>>  
>>        case ::uhd::rx_metadata_t::ERROR_CODE_OVERFLOW:
>>          _tag_now = true;
>> +
>> +        //publish message, only if someone's listening
>> +        if(! pmt::is_null(d_message_subscribers) )
>> +        {
>> +          pmt::pmt_t dic = pmt::make_dict();
>> +          dic = pmt::dict_add(dic, pmt::mp("code"),
>pmt::from_long(_metadata.error_code));
>
>I was trying to avoid the case where the message we send is a
>dictionary with a single entry "code", where the code is "no error".
>
>I'm still not entirely convinced about this anyway. If you're running
>at multiple 100 Msps, you will get a lot of work calls, and a lot of
>consequent messages.
>
>If this was only on:
>- Errors (O, U, S, D)
>- Re-tunes
>- Any time an rx_time gets written
>- Or even a selectable subset of these
>
>That would seem more manageable.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/gnuradio/gnuradio/pull/835/files/bf7ac1bfa972a5bc97f4d485531e82f6fc277e9b#r65974795

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.